model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -302,7 +295,7 @@ data class RootDependencyIndex(
  * Note: This is by intention no data class. Equality is tested via references and not via the values contained.
  */
 @JsonInclude(JsonInclude.Include.NON_DEFAULT)
-class DependencyReference(
+data class DependencyReference(
```

Comment:
This change makes me worry. The class comment says "This is by intention no data class. Equality is tested via references and not via the values contained." If you now have a generated `equals()` implementation that checks all properties, you can get very expensive traversals over whole graphs.

Reply:
Hm, the tests which started failing when stopping to implement `Comparable`, could it be that they asserted a bit too much? (e.g. is there maybe a fix for this possible in test code?)

Reply:
Another alternative could be to hold these dependency references always in lists, not sets, to avoid the `equals()` calls in the first place.

Reply:
`equals()` is called when constructing the dependency graph to match subtrees. This was the reason why this class was not a data class. I am pretty sure that this change will blow up for bigger graphs. Did you test this with a larger project?